dashboard/api_test.py

```diff
@@ -372,7 +383,32 @@ def test_enrolled_upgradable(self):
         run_status = api.get_status_for_courserun(current_run, self.mmtrack)
         assert run_status.status == api.CourseRunStatus.MISSED_DEADLINE
 
-    def test_not_paid_not_passed(self):
+    def test_no_past_present_future(self):
+        """
+        Test in case the course run returns False to all the
+        checks is_current, is_past, is_future, has_frozen_grades
+        """
+        self.mmtrack.configure_mock(**{
+            'is_enrolled.return_value': True,
+            'is_enrolled_mmtrack.return_value': True,
+            'has_paid.return_value': True
+        })
+        crun = self.create_run(
+            start=None,
+            end=None,
+            enr_start=None,
+            enr_end=None,
+            edx_key="course-v1:MITx+8.MechCX+2014_T1"
+        )
+        with self.assertRaises(ImproperlyConfigured):
+            api.get_status_for_courserun(crun, self.mmtrack)
+
+    @ddt.data(
+        (False, None, api.CourseRunStatus.MISSED_DEADLINE),
```

Comment:
I think `@ddt.unpack` looks fine here but I'm wondering about the `None` in `False, None, api.CourseRunStatus.MISSED_DEADLINE` above. Is that intentional? It seems like `has_frozen_grades` should return either `True` or `False`, not `None`